### PR TITLE
Features/limit order bot

### PR DESCRIPTION
**Pull request by webelf101**
## Background

Limit order bot smart contracts in Rust.

### CODE REVIEW

**Comment by taariq** [comment]
@webelf101 & @wc117 What's the delay blocking getting this into production?

File: amm/limit-order-bot/.circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
+version: 2.1
```
Comment (Vizualni) [inline amm/limit-order-bot/.circleci/config.yml:1]: are we using circleci?

File: amm/limit-order-bot/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+cosmwasm-std = "1.0.0"
+cosmwasm-storage = "1.0.0"
+cw-storage-plus = "0.14.0"
+cw2 = "0.14.0"
+schemars = "0.8.10"
+serde = { version = "1.0.140", default-features = false, features = ["derive"] }
+thiserror = { version = "1.0.31" }
+ethabi = "17.1.0"
+eyre = "0.6.8"
+pyth-sdk = "0.4.2"
```
Comment (Vizualni) [inline amm/limit-order-bot/Cargo.toml:52]: I am not sure how rust works when building the code. Does dependency that's defined here (regardless of whether they are being used in the codebase) end up in a compiled binary? Do we need to have all these libraries here? e.g. afaik eyre and thiserror are both working with errors. Do we need both? This is just me wanting to learn more about rust :)

Comment (Vizualni) [reply amm/limit-order-bot/Cargo.toml:52]: cc @measure-fi

Comment (measure-fi) [reply amm/limit-order-bot/Cargo.toml:52]: I would probably prefer one or the other out of `thiserror` and `eyre`, I think `thiserror` was used in the template. There are a couple different mechanisms we enable to strip out unused symbols from generated binaries, so these will only really add to code bloat if we use them.

**Pull request closed without merging**